SimTKcommon/src/Pathname.cpp

```diff
@@ -110,94 +110,117 @@ static void removeDriveInPlace(string& inout, string& drive) {
-        isAbsolutePath = true;
-        processed.replace(0,2,getCurrentWorkingDirectory(drive));
-        removeDriveInPlace(processed, drive);
-    } else if (processed.substr(0,2) == "@/") {
-        isAbsolutePath = true;
-        processed.replace(0,2,getThisExecutableDirectory());
-        removeDriveInPlace(processed, drive);
-    } else if (!drive.empty()) {
-        // Looks like a relative path name. But if it had an initial
-        // drive specification, e.g. X:something.txt, that is supposed
-        // to be interpreted relative to the current working directory
-        // on drive X, just as though it were X:./something.txt.
-        isAbsolutePath = true;
-        processed.insert(0, getCurrentWorkingDirectory(drive));
-        removeDriveInPlace(processed, drive);
+    if (processed.substr(0, 1) == "/") {
+        dontApplySearchPath = true;
+        processed.erase(0, 1);
+        if (!swddrive.empty()) finaldrive = swddrive;
+        else if (!pathdrive.empty()) finaldrive = pathdrive;
+        else finaldrive = getCurrentDriveLetter();
+    }
+    else if (processed.substr(0, 2) == "./") {
+        dontApplySearchPath = true;
+        String discard;
+        // If swd was empty, then "./" came from path. Make sure not to write
+        // over current pathdrive or swddrive.
+        if (swd.empty()) {
+            processed.replace(0, 2, getCurrentWorkingDirectory(pathdrive));
+            removeDriveInPlace(processed, discard);
+        }
+        // Otherwise, "./" came from swd
+        else {
+            processed.replace(0, 2, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(processed, discard);
+        }
+        if (!swddrive.empty()) finaldrive = swddrive;
+        else if (!pathdrive.empty()) finaldrive = pathdrive;
+        else finaldrive = getCurrentDriveLetter();
+    }
+    else if (processed.substr(0, 2) == "@/") {
+        dontApplySearchPath = true;
+        processed.replace(0, 2, getThisExecutableDirectory());
+        removeDriveInPlace(processed, finaldrive);
+    }
+    // Looks like a relative path name. But if either swd or path had 
+    // an initial drive specification, e.g. X:something.txt, that is 
+    // supposed to be interpreted relative to the current working directory
+    // on drive X, just as though it were X:./something.txt.
+    // Check swd first as its drive takes precedence.
+    else if (!swddrive.empty()) {
+        dontApplySearchPath = true;
+        processed.insert(0, getCurrentWorkingDirectory(swddrive));
+        removeDriveInPlace(processed, finaldrive);
+    }
+    else if (!pathdrive.empty()) {
+        dontApplySearchPath = true;
+        // Even if swd did not provide a drive, swd may not have been empty.
+        // Make sure to still use pathdrive.
+        if (!swd.empty()) {
+            processed.insert(0, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(processed, swddrive);
+            finaldrive = pathdrive;
```

Comment:
I confused by this code and the comment above it. Why is `swddrive` being used here when `pathdrive` is the only non-empty one? Is there a test case missing for this?